dataverse-persistence/src/main/java/edu/harvard/iq/dataverse/persistence/dataset/DatasetVersion.java

```diff
@@ -205,6 +205,18 @@ public List<DatasetField> getDatasetFields() {
     public List<DatasetField> getDatasetFieldsOptional() {
         return datasetFieldsOptional;
     }
+    
+    public List<DatasetField> getDatasetFieldsAll() {
+
+        if (getDatasetFieldsOptional().isEmpty()) {
+            return getDatasetFields();
+        } else {
+            final List<DatasetField> result = new ArrayList<DatasetField>(
```

Comment:
What is the point of that final declaration? Seems redundant.
Also this variable seems redundant. Would not be better to inline the return statement? Like:
`getDatasetFields().addAll(getDatasetFieldsOptional())`?

Reply:
Proposed code would modify this.datasetFields list, getters generally are not expected to modify object state.
( https://en.wikipedia.org/wiki/Command%E2%80%93query_separation ).
Wil not fix.